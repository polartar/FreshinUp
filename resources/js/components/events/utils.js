import get from 'lodash/get'

const getFileNameCopy = (name) => {
  const regex = /\s*\(([0-9]+)\)$/gm
  const matches = name.match(regex) || []
  const count = (
    parseInt(
      get(matches, '[0]', '')
        .replace('(', '')
        .replace(')', '')
    ) || 0
  ) + 1
  return `Copy of ${name.replace(get(matches, '[0]', ''), '')} (${count})`
}

export default getFileNameCopy
