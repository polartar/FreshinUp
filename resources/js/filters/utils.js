export function prefixStr (value, prefix = '@') {
  if (!value) return ''
  return value ? `${prefix} ${value}` : ''
}
