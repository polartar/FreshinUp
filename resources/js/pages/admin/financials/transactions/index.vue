<template>
  <div>
    <v-dialog
      v-model="advancedDialog"
      scrollable
      max-width="436"
      content-class="right-dialog"
    >
      <advanced-filter
        v-if="!isLoading"
        :max-width-bottom-row="422"
        :base-filters="basicFilters"
        :advanced-filters="advancedFilters"
        :payment-types="paymentTypes"
        :devices="devices"
        @close="advancedDialog = false"
      />
    </v-dialog>
    <v-layout
      row
      wrap
    >
      <v-flex
        d-inline-flex
        align-center
        ma-2
      >
        <h2>{{ pageTitle }}</h2>
      </v-flex>
      <v-flex
        text-xs-right
      >
        <v-btn
          color="primary"
          dark
          @click.stop="advancedDialog = true"
        >
          Advanced Filters
        </v-btn>
      </v-flex>
    </v-layout>
    <v-divider />

    <br>
    <v-card
      class="pa-4"
    >
      <basic-filter
        v-if="!isLoading"
        :filters="basicFilters"
      />
    </v-card>

    <v-layout
      row
      px-5
      py-2
      wrap
    >
      <v-flex
        xs12
        sm12
        md6
      >
        <v-card class="mx-2">
          <v-card-title class="font-weight-bold subheading">
            Total Sales / Time
          </v-card-title>
          <v-divider />
          <sales-chart
            v-if="!isLoading"
            :sales="financialSummary.sales_time"
          />
        </v-card>
      </v-flex>
      <v-flex
        xs12
        sm12
        md3
      >
        <v-card class="mx-2">
          <v-card-title class="font-weight-bold subheading">
            Total Sales
          </v-card-title>
          <v-divider />
          <total-sales
            v-if="!isLoading"
            :gross="financialSummary.gross"
            :net="financialSummary.net"
            :payment-type-totals="financialSummary.sales_type"
          />
        </v-card>
      </v-flex>
      <v-flex
        xs12
        sm12
        md3
      >
        <v-card class="mx-2">
          <v-card-title class="font-weight-bold subheading">
            AVG. Ticket
          </v-card-title>
          <v-divider />
          <average-ticket
            v-if="!isLoading"
            :average-ticket="financialSummary.avg_ticket"
          />
        </v-card>
        <v-card class="mx-2 mt-2">
          <v-card-title class="font-weight-bold subheading">
            Sales By Method
          </v-card-title>
          <v-divider />
          <sales-by-method
            v-if="!isLoading"
            :payment-type-totals="financialSummary.sales_type"
          />
        </v-card>
      </v-flex>
    </v-layout>
  </div>
</template>

<script>
import { mapGetters, mapActions } from 'vuex'
import AdvancedFilter from '~/components/financials/AdvancedFilter.vue'
import BasicFilter from '~/components/financials/BasicFilter.vue'
import AverageTicket from '~/components/financials/AverageTicket.vue'
import SalesByMethod from '~/components/financials/SalesByMethod.vue'
import SalesChart from '~/components/financials/SalesChart.vue'
import TotalSales from '~/components/financials/TotalSales.vue'
import moment from 'moment'
import { forEach } from 'lodash'

export default {
  layout: 'admin',
  components: {
    AdvancedFilter,
    BasicFilter,
    AverageTicket,
    SalesByMethod,
    SalesChart,
    TotalSales
  },
  data () {
    return {
      pageTitle: 'Financial Reports',
      advancedDialog: false,
      lastFilterParams: {},
      basicFilters: {
        event_uuid: this.$route.query.event_uuid || null,
        company_uuid: this.$route.query.company_uuid || null,
        fleet_member_uuid: this.$route.query.fleet_member_uuid || null,
        contractor_uuid: this.$route.query.contractor_uuid || null,
        date_after: this.$route.query.date_after || null,
        date_before: this.$route.query.date_before || null
      },
      advancedFilters: {
        event_tag_uuid: this.$route.query.event_tag_uuid || null,
        location_uuid: this.$route.query.location_uuid || null,
        customer_uuid: this.$route.query.customer_uuid || null,
        staff_uuid: this.$route.query.staff_uuid || null,
        device_uuid: this.$route.query.device_uuid || null,
        category_uuid: this.$route.query.category_uuid || null,
        item_uuid: this.$route.query.item_uuid || null,
        min_price: Number(this.$route.query.min_price) || null,
        max_price: Number(this.$route.query.max_price) || null,
        payment_type_uuid: this.$route.query.payment_type_uuid || null,
        transaction_uuid: this.$route.query.transaction_uuid || null,
        payment_uuid: this.$route.query.payment_uuid || null
      }
    }
  },
  computed: {
    ...mapGetters('page', ['isLoading']),
    ...mapGetters('devices', { 'devices': 'items' }),
    ...mapGetters('paymentTypes', { paymentTypes: 'items' }),
    ...mapGetters('financialsummary', { financialSummary: 'items' })
  },
  methods: {
    ...mapActions('page', {
      setPageLoading: 'setLoading'
    })
  },
  beforeRouteEnterOrUpdate (vm, to, from, next) {
    vm.setPageLoading(true)
    let filtersForFinancialSummary = {}
    forEach(vm.$route.query, function (value, key) {
      filtersForFinancialSummary[key] = value
    })
    Promise.all([
      vm.$store.dispatch('devices/getItems'),
      vm.$store.dispatch('paymentTypes/getItems'),
      vm.$store.dispatch('financialsummary/setFilters', filtersForFinancialSummary),
      vm.$store.dispatch('financialsummary/getItems')
    ]).then(() => {
      vm.$store.dispatch('page/setLoading', false)
      if (next) next()
    })
  }
}
</script>

<style scoped>
  /deep/ .right-dialog  {
    bottom: 0;
    right: 0;
    position: absolute;
    margin-right: 0px;
    margin-bottom: 0px;
  }
</style>
