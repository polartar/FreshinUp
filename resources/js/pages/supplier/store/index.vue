
<script>
import stores from '~/pages/admin/fleet-members/index.vue'
import { mapGetters } from 'vuex'
const INCLUDE = [
  'tags',
  'addresses',
  'status',
  'owner',
  'type'
]

export default {
  extends: stores,
  computed: {
    ...mapGetters(['currentUser'])
  },
  beforeRouteEnterOrUpdate (vm, to, from, next) {
    vm.setPageLoading(true)
    Promise.all([
      vm.$store.dispatch('stores/setFilters', {
        ...vm.$route.query,
        ...this.lastFilterParams,
        owner_uuid: vm.currentUser.uuid
      }),
      vm.$store.dispatch('storeStatuses/getItems', {
        params: { include: INCLUDE }
      })
    ])
      .then(() => {
        if (next) next()
      })
      .catch((error) => console.error(error))
      .then(() => vm.setPageLoading(false))
  }
}
</script>
