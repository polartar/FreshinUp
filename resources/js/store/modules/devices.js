import makeRestStore from 'fresh-bus/store/utils/makeRestStore'

export default (initialState = {}) => {
  const store = makeRestStore(
    'devices',
    { item: initialState.item, items: initialState.items },
    {
      itemsPath: () => `/foodfleet/devices`,
      itemPath: ({ id }) => `/foodfleet/devices/${id}`
    }
  )

  return {
    namespaced: true,
    ...store
  }
}
