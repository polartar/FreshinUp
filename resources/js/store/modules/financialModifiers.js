import makeRestStore from 'fresh-bus/store/utils/makeRestStore'

export default ({ items, item }) => {
  return makeRestStore('financial-modifiers', { items, item })
}
