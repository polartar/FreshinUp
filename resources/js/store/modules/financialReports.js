import makeRestStore from '@freshinup/core-ui/src/store/utils/makeRestStore'

export default (initialState = {}) => {
  const store = makeRestStore(
    'financialreports',
    { item: initialState.item, items: initialState.items },
    {
      itemsPath: () => `/foodfleet/financial-reports`,
      itemPath: ({ id }) => `/foodfleet/financial-reports/${id}`
    }
  )

  return {
    namespaced: true,
    ...store
  }
}
