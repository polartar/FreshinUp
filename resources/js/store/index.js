import { createStore } from 'fresh-bus/store'

import page from './modules/page'
import devices from './modules/devices'
import financialModifiers from './modules/financialModifiers'
import financialReports from './modules/financialReports'
import paymentTypes from './modules/paymentTypes'
import financialsummary from './modules/financialsummary'
import squares from './modules/squares'
import documents from './modules/documents'
import documentStatuses from './modules/documentStatuses'
import documentTypes from './modules/documentTypes'
import transactions from './modules/transactions'
import companyDetails from './modules/companyDetails'
import eventSummary from './modules/eventSummary'
import eventTypes from './modules/eventTypes'
import eventHistories from './modules/eventHistories'
import venues from './modules/venues'
import stores from './modules/stores'
import storeStatuses from './modules/storeStatuses'
import storeTypes from './modules/storeTypes'
import companies from './modules/companies'

export default (initialState = {}) => {
  return createStore(
    initialState,
    {
      modules: {
        page: page({}),
        devices: devices({}),
        financialModifiers: financialModifiers({}),
        financialReports: financialReports({}),
        paymentTypes: paymentTypes({}),
        financialsummary: financialsummary({}),
        squares: squares({}),
        documents: documents({}),
        documentStatuses: documentStatuses({}),
        documentTypes: documentTypes({}),
        transactions: transactions({}),
        companyDetails: companyDetails({}),
        eventSummary: eventSummary({}),
        eventTypes: eventTypes({}),
        eventHistories: eventHistories({}),
        venues: venues({}),
        stores: stores({}),
        storeStatuses: storeStatuses({}),
        storeTypes: storeTypes({}),
        companies: companies({})
      }
    }
  )
}
