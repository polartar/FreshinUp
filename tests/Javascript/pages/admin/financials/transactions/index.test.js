import { mount, shallowMount } from '@vue/test-utils'
import { createStore } from 'fresh-bus/store'
import { createLocalVue } from 'fresh-bus/tests/utils'
import Component from '~/pages/admin/financials/transactions/index.vue'
import devices from '~/store/modules/devices'
import paymentTypes from '~/store/modules/paymentTypes'
import financialsummary from '~/store/modules/financialsummary'
import transactions from '~/store/modules/transactions'
import financialModifiers from '~/store/modules/financialModifiers'
import financialReports from '~/store/modules/financialReports'
import { FIXTURE_FINANCIAL_SUMMARY } from 'tests/__data__/financialSummary'
import { FIXTURE_TRANSACTIONS } from 'tests/__data__/transactions'
import { FIXTURE_CURRENT_USER } from 'tests/__data__/user'
import { FIXTURE_FINANCIAL_MODIFIERS } from 'tests/__data__/modifiers'
import { FIXTURE_REPORTABLES } from 'tests/__data__/reportables'

describe('Admin Financial Results Page', () => {
  let localVue, mock, store
  describe('Mount', () => {
    beforeEach(() => {
      const vue = createLocalVue({ validation: true, router: true })
      localVue = vue.localVue
      mock = vue.mock
    })
    afterEach(() => {
      mock.restore()
    })
    test('snapshot', done => {
      const vue = createLocalVue({ validation: true })
      localVue = vue.localVue
      mock = vue.mock

      mock.onGet('api/foodfleet/devices')
        .reply(200, {})

      mock.onGet('api/currentUser')
        .reply(200, FIXTURE_CURRENT_USER)

      mock.onGet('api/foodfleet/payment-types')
        .reply(200, {})

      mock.onGet('api/foodfleet/financial-modifiers')
        .reply(200, FIXTURE_FINANCIAL_MODIFIERS)

      mock.onGet('api/foodfleet/financial-summary')
        .reply(200, FIXTURE_FINANCIAL_SUMMARY)

      mock.onGet('api/foodfleet/transactions')
        .reply(200, { data: FIXTURE_TRANSACTIONS })

      mock.onAny().reply(config => {
        console.warn('No mock match for ' + config.url, config)
        return [404, { message: 'No mock match for ' + config.url, data: config }]
      })

      const store = createStore({}, {
        modules: {
          devices: devices({}),
          paymentTypes: paymentTypes({}),
          financialsummary: financialsummary({}),
          transactions: transactions({}),
          financialModifiers: financialModifiers({})
        }
      })

      const wrapper = mount(Component, {
        localVue: localVue,
        store,
        stubs: {
          Doughnut: true,
          Lines: true
        },
        mocks: {
          $route: {
            query: {}
          }
        }
      })

      // Action: load the page data
      Component.beforeRouteEnterOrUpdate(wrapper.vm, null, null, async () => {
        await wrapper.vm.$nextTick()
        expect(wrapper.element).toMatchSnapshot()
        done()
      })
    })
  })
  describe('Methods', () => {
    beforeEach(() => {
      const vue = createLocalVue({ validation: true })
      localVue = vue.localVue
      mock = vue.mock

      mock.onGet('api/foodfleet/devices')
        .reply(200, {})

      mock.onGet('api/currentUser')
        .reply(200, FIXTURE_CURRENT_USER)

      mock.onPut('api/users')
        .reply(201, FIXTURE_CURRENT_USER)

      mock.onGet('api/foodfleet/payment-types')
        .reply(200, {})

      mock.onGet('api/foodfleet/financial-modifiers')
        .reply(200, FIXTURE_FINANCIAL_MODIFIERS)

      mock.onGet('api/foodfleet/financial-summary')
        .reply(200, FIXTURE_FINANCIAL_SUMMARY)

      mock.onGet('api/foodfleet/transactions')
        .reply(200, { data: FIXTURE_TRANSACTIONS })

      mock.onPost('api/foodfleet/financial-reports')
        .reply(201, FIXTURE_REPORTABLES[0])

      mock.onAny().reply(config => {
        console.warn('No mock match for ' + config.url, config)
        return [404, { message: 'No mock match for ' + config.url, data: config }]
      })

      store = createStore({}, {
        modules: {
          devices: devices({}),
          paymentTypes: paymentTypes({}),
          financialsummary: financialsummary({}),
          transactions: transactions({}),
          financialModifiers: financialModifiers({}),
          financialReports: financialReports({})
        }
      })
    })
    afterEach(() => {
      mock.restore()
    })
    test('onPaginate()', async () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store,
        stubs: {
          Doughnut: true,
          Lines: true
        },
        mocks: {
          $route: {
            query: {}
          }
        }
      })
      // Action: change State Machine's state
      await wrapper.vm.onPaginate({ rowsPerPage: 5, page: 2 })
      let request = mock.history.get[mock.history.get.length - 1]
      expect(request.params['page[number]']).toEqual(2)
      expect(request.params['page[size]']).toEqual(5)
    })
    test('onSaveSearch()', async () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store,
        stubs: {
          Doughnut: true,
          Lines: true
        },
        mocks: {
          $route: {
            query: {}
          }
        }
      })
      // Action: change State Machine's state
      await wrapper.vm.onSaveSearch({
        name: 'Test',
        modifier_1_id: 1,
        modifier_2_id: 2
      })
      await wrapper.vm.$nextTick()
      let request = mock.history.post[0]
      expect(request.data).toEqual('{"name":"Test","modifier_1_id":1,"modifier_2_id":2,"filters":"{}"}')
    })
    test('saveParameters()', async () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store,
        stubs: {
          Doughnut: true,
          Lines: true
        },
        mocks: {
          $route: {
            query: {}
          }
        }
      })
      // Action: change State Machine's state
      await wrapper.vm.saveParameters([
        'parameter_1',
        'parameter_1'
      ])
      await wrapper.vm.$nextTick()
      let request = mock.history.put[0]
      expect(request.data).toEqual('{"data_visibility":["parameter_1","parameter_1"]}')
    })
  })
})
