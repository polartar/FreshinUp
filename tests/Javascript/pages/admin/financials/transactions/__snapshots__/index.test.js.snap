// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Financial Results Page Mount snapshot 1`] = `
<div>
  <div
    class="v-dialog__container"
    style="display: block;"
  >
    <div
      class="v-dialog__content"
      style="z-index: 0;"
      tabindex="-1"
    >
      <div
        class="v-dialog right-dialog v-dialog--scrollable"
        style="max-width: 436px; display: none;"
      />
    </div>
  </div>
   
  <div
    class="v-dialog__container"
    style="display: block;"
  >
    <div
      class="v-dialog__content"
      style="z-index: 0;"
      tabindex="-1"
    >
      <div
        class="v-dialog right-dialog v-dialog--scrollable"
        style="max-width: 436px; display: none;"
      />
    </div>
  </div>
   
  <div
    class="v-dialog__container"
    style="display: block;"
  >
    <div
      class="v-dialog__content"
      style="z-index: 0;"
      tabindex="-1"
    >
      <div
        class="v-dialog v-dialog--scrollable"
        style="max-width: 436px; display: none;"
      >
        <div
          class="v-card v-sheet theme--light"
        >
          <div
            class="v-card__title justify-space-between px-4"
          >
            <span
              class="black--text font-weight-bold title text-uppercase"
            >
              Save Report
            </span>
             
            <button
              class="mr-0 mt-2 small-btn v-btn v-btn--icon v-btn--large theme--light"
              type="button"
            >
              <div
                class="v-btn__content"
              >
                <i
                  aria-hidden="true"
                  class="v-icon grey--text far fa-times-circle theme--light"
                />
              </div>
            </button>
          </div>
           
          <hr />
           
          <div
            class="pa-4"
          >
            <form
              class="v-form"
              novalidate="novalidate"
            >
              <div
                class="layout row"
              >
                <div
                  class="flex sm12"
                >
                  <div
                    class="layout row justify-space-between my-2"
                  >
                    
            Title
            
                    <!---->
                  </div>
                   
                  <div
                    class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed theme--light"
                  >
                    <div
                      class="v-input__control"
                    >
                      <div
                        class="v-input__slot"
                      >
                        <div
                          class="v-text-field__slot"
                        >
                          <input
                            data-vv-name="name"
                            required="required"
                            type="text"
                          />
                        </div>
                      </div>
                      <div
                        class="v-text-field__details"
                      >
                        <div
                          class="v-messages theme--light"
                        >
                          <div
                            class="v-messages__wrapper"
                          />
                        </div>
                        <div
                          class="v-counter theme--light"
                        >
                          0 / 255
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
               
              <div
                class="layout row mb-2"
              >
                <div
                  class="flex sm6 pr-2"
                >
                  <div
                    class="layout row justify-space-between my-2"
                  >
                    
            Modifier 1
            
                    <!---->
                  </div>
                   
                  <div
                    class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-select v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light"
                    role="combobox"
                  >
                    <div
                      class="v-input__control"
                    >
                      <div
                        class="v-input__slot"
                      >
                        <div
                          class="v-select__slot"
                        >
                          <div
                            class="v-select__selections"
                          >
                            <div
                              class="v-select__selection v-select__selection--comma"
                            >
                              None selected
                            </div>
                            <input
                              aria-readonly="false"
                              autocomplete="on"
                              readonly="readonly"
                              type="text"
                            />
                          </div>
                          <div
                            class="v-input__append-inner"
                          >
                            <div
                              class="v-input__icon v-input__icon--append"
                            >
                              <i
                                aria-hidden="true"
                                class="v-icon material-icons theme--light"
                              >
                                arrow_drop_down
                              </i>
                            </div>
                          </div>
                        </div>
                        <div
                          class="v-menu"
                        >
                          <div
                            class="v-menu__content theme--light "
                            style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                          >
                            <div
                              class="v-select-list v-card theme--light"
                            >
                              <div
                                class="v-list theme--light"
                                role="list"
                              >
                                <div
                                  class="primary--text"
                                  role="listitem"
                                >
                                  <a
                                    class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text"
                                  >
                                    <div
                                      class="v-list__tile__content"
                                    >
                                      <div
                                        class="v-list__tile__title"
                                      >
                                        None selected
                                      </div>
                                    </div>
                                  </a>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                 
                <div
                  class="flex sm6 pl-2"
                >
                  <div
                    class="layout row justify-space-between my-2"
                  >
                    
            Modifier 2
            
                    <!---->
                  </div>
                   
                  <div
                    class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-select v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light"
                    role="combobox"
                  >
                    <div
                      class="v-input__control"
                    >
                      <div
                        class="v-input__slot"
                      >
                        <div
                          class="v-select__slot"
                        >
                          <div
                            class="v-select__selections"
                          >
                            <div
                              class="v-select__selection v-select__selection--comma"
                            >
                              None selected
                            </div>
                            <input
                              aria-readonly="false"
                              autocomplete="on"
                              readonly="readonly"
                              type="text"
                            />
                          </div>
                          <div
                            class="v-input__append-inner"
                          >
                            <div
                              class="v-input__icon v-input__icon--append"
                            >
                              <i
                                aria-hidden="true"
                                class="v-icon material-icons theme--light"
                              >
                                arrow_drop_down
                              </i>
                            </div>
                          </div>
                        </div>
                        <div
                          class="v-menu"
                        >
                          <div
                            class="v-menu__content theme--light "
                            style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                          >
                            <div
                              class="v-select-list v-card theme--light"
                            >
                              <div
                                class="v-list theme--light"
                                role="list"
                              >
                                <div
                                  class="primary--text"
                                  role="listitem"
                                >
                                  <a
                                    class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text"
                                  >
                                    <div
                                      class="v-list__tile__content"
                                    >
                                      <div
                                        class="v-list__tile__title"
                                      >
                                        None selected
                                      </div>
                                    </div>
                                  </a>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
               
              <div
                class="layout row mt-4 mb-2"
              >
                <div
                  class="flex"
                >
                  <button
                    class="ml-0 v-btn theme--light"
                    type="button"
                  >
                    <div
                      class="v-btn__content"
                    >
                      
            Cancel
          
                    </div>
                  </button>
                </div>
                 
                <div
                  class="flex texs-xs-right"
                >
                  <button
                    class="mr-0 right primary v-btn theme--light"
                    type="button"
                  >
                    <div
                      class="v-btn__content"
                    >
                      
            Save
          
                    </div>
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="v-dialog__container"
    style="display: block;"
  >
    <div
      class="v-dialog__content"
      style="z-index: 0;"
      tabindex="-1"
    >
      <div
        class="v-dialog v-dialog--scrollable"
        style="max-width: 400px; display: none;"
      />
    </div>
  </div>
   
  <div
    class="layout row wrap pt-3"
  >
    <div
      class="flex d-inline-flex align-center ma-2"
    >
      <h2
        class="white--text"
      >
        
        Financial Reports
      
      </h2>
    </div>
     
    <div
      class="flex text-xs-right"
    >
      <button
        class="v-btn theme--dark primary"
        type="button"
      >
        <div
          class="v-btn__content"
        >
          
        Advanced Filters
      
        </div>
      </button>
       
      <button
        class="v-btn theme--dark primary"
        type="button"
      >
        <div
          class="v-btn__content"
        >
          
        Data Visibility
      
        </div>
      </button>
       
      <button
        class="v-btn theme--dark primary"
        type="button"
      >
        <div
          class="v-btn__content"
        >
          
        Save Report
      
        </div>
      </button>
       
      <button
        class="v-btn theme--dark primary"
        type="button"
      >
        <div
          class="v-btn__content"
        >
          
        Export
      
        </div>
      </button>
    </div>
  </div>
   
  <hr
    class="v-divider theme--light"
  />
   
  <br />
   
  <div
    class="pa-4 v-card v-sheet theme--light"
  />
   
  <div
    class="layout row px-5 py-2 wrap"
  >
    <div
      class="flex xs12 sm12 md6"
    >
      <div
        class="mx-2 v-card v-sheet theme--light"
      >
        <div
          class="v-card__title font-weight-bold subheading"
        >
          
          Total Sales / Time
        
        </div>
         
        <hr
          class="v-divider theme--light"
        />
         
        <!---->
      </div>
    </div>
     
    <div
      class="flex xs12 sm12 md3"
    >
      <div
        class="mx-2 v-card v-sheet theme--light"
      >
        <div
          class="v-card__title font-weight-bold subheading"
        >
          
          Total Sales
        
        </div>
         
        <hr
          class="v-divider theme--light"
        />
         
        <!---->
      </div>
    </div>
     
    <div
      class="flex xs12 sm12 md3"
    >
      <div
        class="mx-2 v-card v-sheet theme--light"
      >
        <div
          class="v-card__title font-weight-bold subheading"
        >
          
          AVG. Ticket
        
        </div>
         
        <hr
          class="v-divider theme--light"
        />
         
        <!---->
      </div>
       
      <div
        class="mx-2 mt-2 v-card v-sheet theme--light"
      >
        <div
          class="v-card__title font-weight-bold subheading"
        >
          
          Sales By Method
        
        </div>
         
        <hr
          class="v-divider theme--light"
        />
         
        <!---->
      </div>
    </div>
  </div>
   
  <div
    class="layout row px-5 py-4"
  >
    <div
      class="flex"
    >
      <div
        class="v-card v-sheet theme--light"
      >
        <div
          class="v-card__title"
        >
          <div
            class="layout"
          >
            <div
              class="flex xs12 sm6"
            >
              <p
                class="font-weight-bold mt-3"
              >
                
                Showing 0 Results
              
              </p>
            </div>
             
            <div
              class="flex xs12 sm6"
            >
              <p
                class="mt-3 text-sm-right"
              >
                
                Tip: use the 
                <a
                  class="primary--text underline"
                >
                   Data visibility
                
                </a>
                 settings to list specific parameters
              
              </p>
            </div>
          </div>
        </div>
         
        <!---->
      </div>
    </div>
  </div>
</div>
`;
